cloudinit/sources/helpers/openstack.py

```diff
@@ -672,7 +673,14 @@ def convert_net_json(network_json=None, known_macs=None):
                 if k == "bond_links":
                     continue
                 elif k.startswith("bond"):
-                    params.update({k: v})
+                    # There is a difference in key name formatting for
+                    # bond parameters in the cloudinit and OpenStack
+                    # network schemas. The keys begin with 'bond-' in the
+                    # cloudinit schema but 'bond_' in OpenStack
+                    # network_data.json schema. Translate them to what
+                    # is expected by cloudinit.
+                    translated_key = re.sub(r'^(bond)_(.+)$', r'\1-\2', k)
```

Comment:
Being this a simple split, could do something in the lines of the following and avoid using `re` to reduce complexity:

```python
translated_key = k.split("bond_", 1)[-1]
```

Reply:
Thanks for the suggestion. Updated as requested to remove `re`.